sway-core/src/semantic_analysis/type_check_context.rs

```diff
@@ -71,6 +74,7 @@ impl<'ns> TypeCheckContext<'ns> {
             self_type: insert_type(TypeInfo::Contract),
             mode: Mode::NonAbi,
             purity: Purity::default(),
+            kind: None,
```

Comment:
```suggestion
            kind: TreeType::Contract,
```

Would you change this to `TreeType::Contract` to follow `self_type: insert_type(TypeInfo::Contract),`?